docs/samples/browser-plugin-meetings/app.js

```diff
@@ -1206,53 +1171,38 @@ async function startScreenShare() {
   // Using async/await to make code more readable
   console.log('MeetingControls#startScreenShare()');
   try {
-    let localShareVideoTrack, localShareAudioTrack;
-    if (isMultistream) {
-      [localShareVideoTrack, localShareAudioTrack] = await webex.meetings.mediaHelpers.createDisplayTrackWithAudio();
-    } else {
-      localShareVideoTrack = await webex.meetings.mediaHelpers.createDisplayTrack();
-    }
+    //TODO: Add audio share toggle in sample app
+    const [localShareVideoStream, localShareAudioStream] = await webex.meetings.mediaHelpers.createDisplayStreamWithAudio();
 
-    if (localShareVideoTrack) {
-      localMedia.screenShare.video = localShareVideoTrack;
-      localMedia.screenShare.video.on('published-state-update', ({isPublished}) => {
-        if (!isPublished) {
-          console.log('MeetingControls#startScreenShare() :: local share video track unpublished, stopping it');
+    localMedia.screenShare.video = localShareVideoStream;
+    let isAudioShareEnded = false,
+        isVideoShareEnded = false;
+    localMedia.screenShare.video.on('stream-ended', () => {
+      console.log('MeetingControls#startScreenShare() :: local share video stream ended');
 
-          localMedia.screenShare.video.stop();
-          localMedia.screenShare.video = undefined;
+      localMedia.screenShare.video = undefined;
+      isVideoShareEnded = true;
 
-          meetingStreamsLocalShareVideo.srcObject = null;
-        }
-      });
+      if(isAudioShareEnded && isVideoShareEnded) meetingStreamsLocalShare.srcObject = null;
+    });
 
-      meetingStreamsLocalShareVideo.srcObject = localShareVideoTrack.underlyingStream;
+    localMedia.screenShare.audio = localShareAudioStream;
+    localMedia.screenShare.audio.on('stream-ended', () => {
+      console.log('MeetingControls#startScreenShare() :: local share audio stream ended');
 
-      console.log('MeetingControls#startScreenShare() :: publishing share video track');
-    }
-    
-    if (localShareAudioTrack) {
-      localMedia.screenShare.audio = localShareAudioTrack;
-      localMedia.screenShare.audio.on('published-state-update', ({isPublished}) => {
-        if (!isPublished) {
-          console.log('MeetingControls#startScreenShare() :: local share audio track unpublished, stopping it');
+      localMedia.screenShare.audio = undefined;
+      isAudioShareEnded = true;
 
-          localMedia.screenShare.audio.stop();
-          localMedia.screenShare.audio = undefined;
+      if(isAudioShareEnded && isVideoShareEnded) meetingStreamsLocalShare.srcObject = null;
```

Comment:
why do we need `isAudioShareEnded` and `isVideoShareEnded`? we can do this by just checking `localMedia.screenShare.audio` and `localMedia.screenShare.video`, right?

Reply:
I agree. I introduced that so, both events will update their respective variables and one of them would make the srcObject to null but it makes sense to use `localMedia` reference instead. Will update it.

Reply:
During my published-state-update event PR, I noticed these errors while testing in the samples app. I have updated some of these code as per the error and also checked the discussion above. Below is the PR with the changes: https://github.com/webex/webex-js-sdk/pull/3052

Will comment the PR link wherever I have already rectified it.

Reply:
Thanks @meta-dipanshu-sharma